src/rttr/detail/base/core_prerequisites.h

```diff
@@ -62,6 +62,10 @@ namespace rttr
 // Compiler
 /////////////////////////////////////////////////////////////////////////////////////////
 #if defined( __clang__ )
+
+#if defined __apple_build_version__
```

Comment:
Isn't the apple clang compiler because of its own version scheme,
a compiler for its own?
Because of that I would suggest, you tread him like an own compiler.
```
#if defined(__apple_build_version__)
#   define RTTR_COMPILER RTTR_COMPILER_APPLE_CLANG
```

Reply:
jeah, I was think so too, but its still clang, an should behave like one, so there I decided, only to have a different version number, but jeah, maybe better to have it as an own compiler